x/btcstaking/types/validate_parsed_message.go

```diff
@@ -129,18 +131,25 @@ func ValidateParsedMessageAgainstTheParams(
 		net,
 	)
 	if err != nil {
-		return nil, ErrInvalidUnbondingTx.Wrapf("err: %v", err)
+		return nil, ErrInvalidUnbondingTx.Wrapf("failed to rebuid the unbonding info: %v", err)
 	}
 
-	unbondingOutputIdx, err := bbn.GetOutputIdxInBTCTx(pm.UnbondingTx.Transaction, unbondingInfo.UnbondingOutput)
-	if err != nil {
-		return nil, ErrInvalidUnbondingTx.Wrapf("unbonding tx does not contain expected unbonding output")
+	unbondingTx := pm.UnbondingTx.Transaction
+	if !bytes.Equal(unbondingTx.TxOut[0].PkScript, unbondingInfo.UnbondingOutput.PkScript) {
+		return nil, ErrInvalidUnbondingTx.
+			Wrapf("the unbonding output script is not expected, expected: %x, got: %s",
+				unbondingInfo.UnbondingOutput.PkScript, unbondingTx.TxOut[0].PkScript)
+	}
+	if unbondingTx.TxOut[0].Value != unbondingInfo.UnbondingOutput.Value {
+		return nil, ErrInvalidUnbondingTx.
+			Wrapf(" the unbonding output value is not expected, expected: %d, got: %d",
+				unbondingTx.TxOut[0].Value, unbondingInfo.UnbondingOutput.Value)
 	}
 
 	err = btcstaking.CheckSlashingTxMatchFundingTx(
 		pm.UnbondingSlashingTx.Transaction,
 		pm.UnbondingTx.Transaction,
-		unbondingOutputIdx,
+		0, // unbonding output always has only 1 output
```

Comment:
hmm if we are hardcoidng this, I feel than we should have a test case to assert that unbonding tx with more that `1` output are rejcected. Maybe in `validate_parsed_message_test.go`

Reply:
good point. Added this case in [e2916de](https://github.com/babylonlabs-io/babylon/pull/249/commits/e2916deecf59750d0c7313cc9f803af59216cd1f)